### DISCUSSION TITLE
FreeRDP behaviour while connecting again to the same host (multiple sessions issue)

### DISCUSSION THREAD

**Discussion by spalek1293** [FreeRDP/FreeRDP#8239] [Q&A]
Hi,

I have question related to FreeRDP behaviour while connecting to Windows Server (2022 in this case) via RD Connection broker.

I have following setup:

1. RDS_Host1 and RDS_Host2 - both hosting same application
2. Connection broker (HA setup) - RDS_CB
3. Raspberry with XFreeRDP 2.8.0

Connections are made to RDS_CB (Session Collection do have RDS_Host1 and RDS_Host2 as RDS pool hosts) from Raspberry. They go via internal network, no gateway is engaged here.

I would like to confirm if what I am seeing is correct or incorrect.

- Raspberry while connecting , will land on RDS_Host1 or RDS_Host2
- If for some reason my XFreeRDP process will crash , I will try to start the connection again
- I may end up on RDS_Host1 or RDS_Host2, but previous session which is still active, isn't respected and 'reconnected' (picked up), but completely new session is established

Same time, if I will manually start the same XFreeRDP command few times from the same Raspberry, using the same RDP Username/password, I will end up having multiple parallel sessions active on Raspberry and Windows Servers too.
That is different behaviour comparing to MS RDP Clinet (mstsc), which would pick up existing session of same user, rather than establish completley new one.

My problem is that I do not want new sessions to be established, but I would like XFreeRDP to pickup existing session for this user.
(several sessions of the same user on same host are causing other problems unfortunately).

Is what I am experiencing - expected behaviour of XFreeRDP ?
Is there a way to tell XFreeRDP to pickup existing session, rather than create a completley new one ?

If anything from above is unclear or requires additional explanation - let me know please.

Lukasz

**Comment by akallabeth** [comment]
the question you have can not be answered as session resumption is decided server side.
I can tell that the `mstsc` behavior is the same as `xfreerdp` on our test setup.

* So, maybe you use something that prevents session identification (e.g. incompatible command line options?) so the server thinks a new client connects?
* Maybe the server has some rule for `mstsc` in place (and not `xfreerdp` or vice versa) for that case?

**Reply by spalek1293** [reply]
Hi Akallabeth

Thanks for responding.

Thanks for corrective wording - session resumption , that's the word I've been missing.

With regards to your additional comments:

- That is what I am trying to find out currently too - command I am using to connect is following:
/usr/bin/xfreerdp +auto-reconnect /auto-reconnect-max-retries:30 /t:'Application 2' /scale-device:100 /v:ILB-CB.COMPANYDOMAIN.LOCAL /load-balance-info:'tsv://MS Terminal Services Plugin.1.RDS-FARM' /shell:'||App2' /u:username /d:domanname.local /p:password /cert-ignore /f

- Unless this is Microsoft standard config , there is nothing (I am sure of it) what would drive different behavior for mstsc or non-mstsc clients connecting. I know GPOs and these configs by heart. 

Can any of my connection settings be misleading destination servers to allow session resumption?
Is session resumption in general expected behavior and what I am seeing is incorrect?
Can any of xfreerdp parameters help destination servers to recognize this device /user connecting again to allow resumption of the session?

I am unsure if my expectation for xfreerdp is correct, but I was hoping for similar behavior as mstsc (session resumption).

Cheers,
Lukasz

**Reply by akallabeth** [reply]
Ok, the `/shell` argument specifies a desktop session with an alternate shell (instead of `explorer.exe`)
I think you meant to use `/app`?
